extensions/sql-database-projects/src/common/httpClient.ts

```diff
@@ -48,4 +54,54 @@ export class HttpClient {
 		}
 		return response.data;
 	}
+
+	/**
+	 * Gets a file/fileContents at the given URL.
+	 * @param downloadUrl The URL to download the file from
+	 * @param targetPath The path to download the file to
+	 * @param outputChannel The output channel to output status messages to
+	 * @returns Full path to the downloaded file or the contents of the file at the given downloadUrl
+	 */
+	public download(downloadUrl: string, targetPath: string, outputChannel?: vscode.OutputChannel): Promise<void> {
```

Comment:
pretty much copied from the Machine Learning extension HttpClient https://github.com/microsoft/azuredatastudio/blob/04af41c4240fe1ea9c6add7dcd23721c05fd22d9/extensions/machine-learning/src/common/httpClient.ts#L38